File: GXpert/GXpert.Web/Modules/Skills/Skill/SkillColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Skills.Columns;

[ColumnsScript("Skills.Skill")]
[BasedOnRow(typeof(SkillRow), CheckNames = true)]
public class SkillColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public string SkillCategoryName { get; set; }
   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Skills/SkillColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SkillRow } from "./SkillRow";

export interface SkillColumns {
    Id: Column<SkillRow>;
    Name: Column<SkillRow>;
    Description: Column<SkillRow>;
    SkillCategoryName: Column<SkillRow>;
}

export class SkillColumns extends ColumnsBase<SkillRow> {
    static readonly columnsKey = 'Skills.Skill';
    static readonly Fields = fieldsProxy<SkillColumns>();
}